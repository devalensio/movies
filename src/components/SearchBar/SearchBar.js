import React, { useState } from 'react';

import styles from './SearchBar.module.scss';

const SearchBar = ({ searchMovie }) => {
  const [searchInput, setSearchInput] = useState('');

  const handleSubmit = (e) => {
    e.preventDefault();

    if (searchInput) searchMovie(searchInput);
  }

  return (
    <form onSubmit={handleSubmit} className={styles.search__bar}>
      <div className={styles['search__bar--main']}>
        <img src="/images/search.png" alt="search-icon" width="16px" height="16px" />
        <input type="text" placeholder="Name of a movie" onChange={(e) => setSearchInput(e.target.value)} />
      </div>
      <button type="submit">Search</button>
    </form>
  );
};

export default SearchBar;
