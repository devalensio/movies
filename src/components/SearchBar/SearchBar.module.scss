@import "../../styles/tokens/all";

.search__bar {
  display: flex;
  &--main {
    width: 70%;
    display: flex;
    border: 1px solid $color-outline-grey;
    border-right: unset;
    padding: $space--xs;
    border-top-left-radius: $space--2xs;
    border-bottom-left-radius: $space--2xs;
    margin-bottom: $space--md;
    input {
      width: 100%;
      margin-left: $space--xs;
      border: unset;
      outline: unset;
    }
  }
  button {
    width: 30%;
    height: 35px;
    border-top-right-radius: $space--2xs;
    border-bottom-right-radius: $space--2xs;
    border: none;
    background-color: $brand-primary-color;
    color: $color--white;
    outline: unset;
    cursor: pointer;
    &:active, &:hover {
      opacity: 0.8;
    }
  }
}
