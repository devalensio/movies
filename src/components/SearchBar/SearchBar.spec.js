import React from 'react';

import renderer from 'react-test-renderer';

import SearchBar from './SearchBar';

describe('Search Bar', () => {
  it('renders SearchBar successfully', () => {
    const component = renderer.create(<SearchBar />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
