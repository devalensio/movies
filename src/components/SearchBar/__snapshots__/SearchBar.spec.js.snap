// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Bar renders SearchBar successfully 1`] = `
<form
  className="search__bar"
  onSubmit={[Function]}
>
  <div
    className="search__bar--main"
  >
    <img
      alt="search-icon"
      height="16px"
      src="/images/search.png"
      width="16px"
    />
    <input
      onChange={[Function]}
      placeholder="Name of a movie"
      type="text"
    />
  </div>
  <button
    type="submit"
  >
    Search
  </button>
</form>
`;
